File: medium/852.Peak Index in a Mountain Array.cpp
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& arr) {
        if(arr.size()==3) return 1;
        int peak,s=0,e=arr.size()-1;
        int mid=s+(e-s)/2;
        while(s<e){
            if(arr[mid]<arr[mid+1]) s=mid+1;
            else e=mid;
            mid=s+(e-s)/2;
        }
        return s;
    }
};

File: medium/852.Peak Index in a Mountain Array.py
class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
        s=0
        e=len(arr)-1
        if e==2:
            return e-1
        while(s<e):
            mid=int(s+(e-s)/2)
            if arr[mid]>arr[mid-1] and arr[mid+1]<arr[mid]:
                return mid
            if arr[mid]>arr[mid-1]:
                s=mid
            else:
                e=mid
        return s # just to adjust syntax
